Source/Libraries/openXDA.APIAuthentication/APIConsoleController.cs

```diff
@@ -178,9 +180,20 @@ public class ConsoleMessage
         [Route("Connect"), HttpGet]
         public IHttpActionResult Connect()
         {
-            ConsoleState connection = new ConsoleState(User, Host);
-            s_activeConnections.Add(connection.SessionID, connection);
-            return Ok(connection.SessionID);
+            try
+            {
+                ConsoleState connection = new ConsoleState(User, Host);
+                s_activeConnections.TryAdd(connection.SessionID, connection);
+                return Ok(connection.SessionID);
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                return Unauthorized();
+            }
+            catch (Exception ex)
+            {
+                return InternalServerError(ex);
+            }
```

Comment:
The reason I added this is becasue we have a `catch (AuthenticationException ex)` that returns a `401` instead of a `500` if we catch a specific exception Type will it still throw the more general exception?